sitetree/utils.py

```diff
@@ -118,37 +152,10 @@ def item(
     item_obj.is_dynamic = True
     item_obj.dynamic_children = []
 
-    cleaned_permissions = []
-    if access_by_perms:
-        # Make permissions a list if currently a single object
-        if not isinstance(access_by_perms, list):
-            access_by_perms = [access_by_perms]
-
-        for perm in access_by_perms:
-            if isinstance(perm, str):
-                # Get permission object from string
-                try:
-                    app, codename = perm.split('.')
-                except ValueError:
-                    raise ValueError(
-                        f'Wrong permission string format: supplied - `{perm}`; '
-                        'expected - `<app_name>.<permission_name>`.')
-
-                try:
-                    perm = Permission.objects.get(codename=codename, content_type__app_label=app)
-
-                except Permission.DoesNotExist:
-                    raise ValueError(f'Permission `{app}.{codename}` does not exist.')
-
-            elif not isinstance(perm, (int, Permission)):
-                raise ValueError('Permissions must be given as strings, ints, or `Permission` instances.')
-
-            cleaned_permissions.append(perm)
-
-    item_obj.permissions = cleaned_permissions or []
+    item_obj.permissions = SimpleLazyObject(lambda: clean_permissions(access_by_perms))
```

Comment:
Perhaps is better to return a `SimpleLazyObject` from within `clean_permissions`. The string parsing/checking etc. isn't a heavy calculation so can happen right away (with the advantage of having feedback directly when starting the server).